ironfish/src/rpc/adapters/tests/adapter.test.ts

```diff
@@ -0,0 +1,61 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import * as yup from 'yup'
+import { FullNode } from '../../../node'
+import { IRpcAdapter, Router, RpcClient } from '../../../rpc'
+import { IronfishSdk } from '../../../sdk'
+import { getUniqueTestDataDir } from '../../../testUtilities'
+
+// eslint-disable-next-line jest/no-export
+export function createAdapterTest(
```

Comment:
nit: this doesn't only create test, also runs it

Reply:
It only creates tests by calling it() that are later discovered by the jest test runner. It doesn't run them.

Look at this example and see how the tests are not detected in the adapterTest.ts. This is because the function doesn't run the test, just create and register tests with the jest runner.

```sh
$ yarn test adapters/tests
 PASS   @ironfish/sdk  src/rpc/adapters/tests/ipc.adapter.test.ts (6.74 s)
 PASS   @ironfish/sdk  src/rpc/adapters/tests/http.adapter.test.ts (6.74 s)
 PASS   @ironfish/sdk  src/rpc/adapters/tests/tcp.adapter.test.ts (7.137 s)

Test Suites: 3 passed, 3 total
Tests:       21 passed, 21 total
Snapshots:   0 total
Time:        7.294 s
```

Reply:
Oh interesting, looks like I need to read about how the test runner picks up tests at runtime.